src/panels/config/automation/condition/types/ha-automation-condition-time.ts

```diff
@@ -36,10 +61,73 @@ export class HaTimeCondition extends LitElement implements ConditionElement {
         .value=${before}
         @value-changed=${this._valueChanged}
       ></paper-input>
+
+      ${Object.keys(DAYS).map(
+        (day) => html`
+          <ha-formfield
+            alignEnd
+            spaceBetween
+            class="flex"
+            .label=${this.hass!.localize(
+              `ui.panel.config.automation.editor.conditions.type.time.weekdays.${day}`
+            )}
+            .dir=${computeRTLDirection(this.hass!)}
+          >
+            <ha-switch
+              .day=${day}
+              .checked=${weekday?.includes(day)}
+              @change=${this._dayValueChanged}
+            >
+            </ha-switch>
+          </ha-formfield>
+        `
+      )}
     `;
   }
 
   private _valueChanged(ev: CustomEvent): void {
     handleChangeEvent(this, ev);
   }
+
+  private _dayValueChanged(ev: CustomEvent): void {
+    ev.stopPropagation();
+
+    const daySwitch = ev.currentTarget as WeekdayHaSwitch;
+
+    let days = this.condition.weekday;
+    if (!days) {
+      days = [];
+    }
+
+    if (daySwitch.checked) {
+      days.push(daySwitch.day);
+    } else {
+      days = days.filter((d) => d !== daySwitch.day);
+    }
+
+    days.sort((a: string, b: string) => DAYS[a] - DAYS[b]);
+
+    this.condition.weekday = days;
+
+    fireEvent(this, "value-changed", {
+      value: this.condition,
+    });
+  }
+
+  static get styles(): CSSResult[] {
+    return [
+      css`
+        .flex {
+          display: flex;
+          height: 40px;
+        }
+      `,
+    ];
+  }
+}
```

Comment:
```suggestion
  static get styles(): CSSResult {
    return css`
        .flex {
          display: flex;
          height: 40px;
        }
      `;
  }
}
```

This may not pass prettier

Reply:
fixed